### PR TITLE
Upgrade gradle and improve release workflow

### PR DESCRIPTION
**Pull request by aymeric-dispa**
Update relelease process: 
- A jar file will be attached automatically once we create a release on Github.
- Gradlew upgrade

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip
 zipStoreBase=GRADLE_USER_HOME
-zipStorePath=wrapper/dists
+zipStorePath=wrapper/dists
```
Comment (ptiurin) [inline gradle/wrapper/gradle-wrapper.properties:5]: not an issue, but it's generally a good practice to leave the newline in. Makes parsing easier.

File: gradlew [hunk 2]
```diff
@@ -205,6 +205,12 @@ set -- \
         org.gradle.wrapper.GradleWrapperMain \
         "$@"
 
+# Stop when "xargs" is not available.
+if ! command -v xargs >/dev/null 2>&1
+then
+    die "xargs is not available"
+fi
+
```
Comment (ptiurin) [inline gradlew:213]: When does this happen? I thought xargs comes by default with most unix systems.

**Review by ptiurin — APPROVED** [review]
Couple of questions, apart from that lgtm.

File: gradlew (hunk 2 above)
Comment (aymeric-dispa) [reply gradlew:213]: This file was generated when I ran command to upgrade gradle

**Pull request merged**